shcDecoder/src/androidTest/java/ca/bc/gov/shcdecoder/SHCVerifierImplTest.kt

```diff
@@ -0,0 +1,94 @@
+        sut.hasValidSignature(FORGED_SHC_URI)
+    }
+
+    @Test
+    fun givenGetImmunizationRecord_whenUserFullyVaccinated_thenShowsFullyImmunized(): Unit = runBlocking {
+        val result = sut.getImmunizationRecord(VALID_FULLY_IMMUNIZED_SHC_URI)
+        Assert.assertEquals(result, testRecord.copy(
+            status = ImmunizationStatus.FULLY_IMMUNIZED
+        ))
+    }
+
+    @Test
+    fun givenGetImmunizationRecord_whenUserPartiallyVaccinated_thenShowsPartiallyImmunized(): Unit = runBlocking {
+        val result = sut.getImmunizationRecord(VALID_PARTIALLY_IMMUNIZED_SHC_URI)
+        Assert.assertEquals(result, testRecord.copy(
+            status = ImmunizationStatus.PARTIALLY_IMMUNIZED
+        ))
+    }
+
+    @Test
+    fun givenGetImmunizationRecord_whenUserNotVaccinated_thenShowsInvalidQrCode(): Unit = runBlocking {
+        val result = sut.getImmunizationRecord(VALID_NO_VACCINES_SHC_URI)
+        Assert.assertEquals(result, testRecord)
+    }
+
+    @Test
+    fun givenGetImmunizationRecord_whenUserHasExempt_thenShowsFullyImmunized(): Unit = runBlocking {
+        val result = sut.getImmunizationRecord(VALID_EXEMPT_SHC_URI)
+        Assert.assertEquals(result, testRecord.copy(
+            status = ImmunizationStatus.FULLY_IMMUNIZED
+        ))
+    }
+
+    companion object {
+        private const val VALID_FULLY_IMMUNIZED_SHC_URI = "shc:/5676290952432060346029243740446031222959532654603460292540772804336028702864716745222809286133314564376531415906402203064504590856435503414245413640370636654171372412363803043756220467374075323239254334433260573601104128125312707425357469745673363745252404076371614204366629033029240527745835041226387507445520406869553635001261102939672426386742702531683972107103555636000553682438630372754325762839456421225824680926317036044040334408766112280634102908097663033508055022776568664444323766396827723059290960005076242528060670383469440844625306597406080565236929557576676121686311763064555721522853413832057032032035394162660831095668595024324562685605407376405762620725761077322156726303505623533936632330557558035609645606230940550560583566712652046675325607200054624540276953120977442027576340760726277761352043617033112163636709225860615530702104547239557667624275681131340321337229401242302404502964203924456342566307284339253505692043757266413557505752657577337429441105543709272661302824364559772043444367336255314161372937425769586465623272625345063774672724286254066609572656554372433410076940563725404000220745742654645320660408226035264025097474047253307130743126370739047541684050300443666976245254703427563158455034676821716361706631324556043924272971110753116933256055270845317137665554761257117535566373094064503553724100103569397228557110751268566125500930572535057250325660564006395374016252660645212843592075501033007006357640103269773827707035385327442245370076773240693939123468412500574144555039656944755544436943343838565225521200377400760061653506263374"
```

Comment:
@JosephFreshworks 
Is it reasonable to create a helper class for all mock data for the test?

Reply:
Could be, let me do it